csharp/src/Ice/Communicator.cs

```diff
@@ -619,20 +620,20 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
                         nameof(caCertificates));
                 }
 
-                _sslEngine = new SslEngine(
+                SslEngine = new SslEngine(
                     this,
                     certificates,
                     caCertificates,
                     certificateSelectionCallback,
                     certificateValidationCallback,
                     passwordCallback);
 
-                IceAddEndpointFactory(Transport.TCP, "tcp", new TcpEndpointFactory(this));
-                IceAddEndpointFactory(Transport.UDP, "udp", new UdpEndpointFactory(this));
-                IceAddEndpointFactory(Transport.WS, "ws", new WSEndpointFactory(this));
-
-                IceAddEndpointFactory(Transport.SSL, "ssl", new SslEndpointFactory(this, _sslEngine));
-                IceAddEndpointFactory(Transport.WSS, "wss", new WSSEndpointFactory(this, _sslEngine));
+                _endpointFactory = new EndpointFactory(this);
```

Comment:
Why is a data member the communicator, afaict it's only useful here.

Reply:
You are right, I remove it.